data_loader.py

```diff
@@ -0,0 +1,246 @@
+##########################
+#Classes to load NEON data
+##########################
+
+import os
+import pathlib
+
+import pandas as pd
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import rasterstats as rs
+import xarray as xr
+import rioxarray as rxr
+import seaborn as sns
+
+class NEONDataLoader:
+    """
+    Parent class to load NEON height data and calculate mean and max
+            
+        Attributes
+        ----------
+        base_dir_tmpl: str
+            template for path to base directory
+        
+        insitu_path_tmpl:
+            template for path to insitu directory
+        
+        chm_path_tmpl: str
+            template for path to lidar directory
+        
+        plots_path_tmpl:
+            template for path to plots directory
+        
+        site_name: str
+            name of the site
+    
+        id_col_name: str
+            name of the ID column header
+    
+        formatting_dict: str
+            specific formatting for the site paths
+    
+        id_modifier: function
+            function to modify the values in the id column
+            
+    """
+    # Set as attributes
+    base_dir_tmpl = os.path.join(
+        'california',
+        'neon-{site_name_low}-site')  
+    insitu_path_tmpl = os.path.join(
+        '{base_dir}',
+        '2013',
+        'insitu',
+        'veg{separator}structure',
+        'D17_2013_{site_name_up}_vegStr.csv')
+    chm_path_tmpl = os.path.join(
+        '{base_dir}',
+        '2013',
+        'lidar',
+        '{site_name_up}_lidarCHM.tif')
+    plots_path_tmpl = os.path.join(
+        '{base_dir}',
+        'vector_data',    
+        '{site_name_up}{plot}_centroids.shp')
+    
+    site_name = NotImplemented
+    id_col_name = NotImplemented
+    formatting_dict = NotImplemented
+    id_modifier = None
+    
+    def __init__(self):
+        # set arguments as attributes
+        self.formatting_dict['site_name_low'] = self.site_name.lower()
+        self.formatting_dict['site_name_up'] = self.site_name.upper()
+        self.formatting_dict['base_dir'] = (
+            self.base_dir_tmpl.format(**self.formatting_dict))
+       
+        self.insitu_path = self.insitu_path_tmpl.format(**self.formatting_dict)
+        self.chm_path = self.chm_path_tmpl.format(**self.formatting_dict)
+        self.plots_path = self.plots_path_tmpl.format(**self.formatting_dict)
+        
```

Comment:
I think you're asking about the 3 dfs below that begin with self._text? If so, great question bc I copied the convention Elsa used in lecture, but I didn't fully understand it. I just looked it up and apparently using "self._text" creates the dfs in the instance of the class, and they exist until the class is destroyed. Honestly, I'm unclear why we need to define them here and set to none, rather than using a normal function below. I need to follow up in office hours to completely understand